bin/main.rb

```diff
@@ -1,60 +1,79 @@
 #!/usr/bin/env ruby
+# rubocop:disable Metrics/BlockLength
 
-# rubocop:disable Lint/UselessAssignment
+require_relative '../lib/game_logic'
+require_relative '../lib/player'
+
+# Initialize the game and two players instances
+game = TicTacToe.new
+player1 = Player.new
+player2 = Player.new
 
 # 1. Prompt both users to give their names
 puts "Player 1, what's your name?"
-player1 = gets.chomp
+player1.name = gets.chomp
 puts '=============================='
 puts "Awesome! Player 2, what's your name?"
-player2 = gets.chomp
+player2.name = gets.chomp
 puts '=============================='
-puts "Welcome #{player1} and #{player2}"
+puts "Welcome #{player1.name} and #{player2.name}"
+
+# Ask Users if they want to see game instructions.
 puts 'Do you want to see game instructions? [y/n]'
 response = gets.chomp
-# 2. If yes, print game instructions! If not, proceed with the following step.
 puts 'Instructions' if response == 'y'
 
-# 3. Define a dummy winning condition (NOT the real game logic)
-winner = false
-
-def announce_winner(player)
-  # This function declares the winner
-
-  puts "#{player}, Congratulations! You won."
-end
-
-# 4. Repeatedly ask the players to make moves till the win/draw condition is met
 rounds = 0
 loop do
-  winner = true if rounds == 3
-
-  board = "
+  puts rounds
+  puts "
   _________________
- |__a__|__b__|__c__|
- |__d__|__e__|__f__|
- |__g__|__h__|__i__|
+ |__#{game.board[0]}__|__#{game.board[1]}__|__#{game.board[2]}__|
+ |__#{game.board[3]}__|__#{game.board[4]}__|__#{game.board[5]}__|
+ |__#{game.board[6]}__|__#{game.board[7]}__|__#{game.board[8]}__|
```

Comment:
- [ ] Since this is some functionality that you might need in different places, could be useful to store in a function. You could have a method that returns this string (in the TicTacToe class) and just print that method here (to respect the DRY principle).